spec/models/manageiq/providers/vmware/infra_manager/provision/cloning_spec.rb

```diff
@@ -1,51 +1,77 @@
-      @vm_prov = FactoryBot.create(:miq_provision_vmware, :userid => @admin.userid, :miq_request => @pr, :source => @vm_template, :request_type => 'template', :state => 'pending', :status => 'Ok', :options => @options)
-    end
+    let(:folder_name)      { 'folder_one' }
+    let(:ems_folder)       { FactoryBot.create(:vmware_folder_vm) }
+    let(:dest_host)        { FactoryBot.create(:host_vmware, :ext_management_system => ems) }
+    let(:dc_nested)        { EvmSpecHelper::EmsMetadataHelper.vmware_nested_folders(ems) }
+    let(:dest_host_nested) { FactoryBot.create(:host_vmware, :ext_management_system => ems).tap { |h| h.parent = dc_nested } }
+    let(:vm_folder_nested) { FactoryBot.create(:ems_folder, :name => 'vm', :ems_id => ems.id).tap { |v| v.parent = dc_nested } }
 
-    let(:folder_name) { 'folder_one' }
-    let(:ems_folder)  { double('ems_folder') }
-    let(:dest_host) do
-      FactoryBot.create(:host_vmware, :ext_management_system => ems)
+    it "returns a folder if one is found" do
+      options[:placement_folder_name] = [ems_folder.id, ems_folder.name]
+      expect(provision).to receive(:find_folder).never
+      provision.dest_folder
     end
 
-    let(:dc_nested) do
-      EvmSpecHelper::EmsMetadataHelper.vmware_nested_folders(@ems)
+    it "attempts to find a usable folder if the ems_folder does not exist" do
+      provision.options[:dest_host] = [dest_host_nested.id, dest_host_nested.name]
+      expect(provision).to receive(:find_folder).once
+      provision.dest_folder
     end
+  end
 
-    let(:dest_host_nested) do
-      FactoryBot.create(:host_vmware, :ext_management_system => @ems).tap { |h| h.parent = dc_nested }
-    end
+  context "#start_clone" do
+    let(:folder)    { FactoryBot.create(:vmware_folder_vm, :ext_management_system => ems) }
+    let(:host)      { FactoryBot.create(:host_vmware_esx, :ext_management_system => ems, :ems_ref => "host-1") }
+    let(:pool)      { FactoryBot.create(:resource_pool_vmware, :ext_management_system => ems) }
+    let(:snapshot)  { FactoryBot.create(:snapshot, :ems_ref => "snapshot-1", :ems_ref_type => "Snapshot") }
+    let(:datastore) { FactoryBot.create(:storage_vmware, :ext_management_system => ems) }
 
-    let(:vm_folder_nested) do
-      FactoryBot.create(:ems_folder, :name => 'vm', :ems_id => @ems.id).tap { |v| v.parent = dc_nested }
+    let(:clone_options) do
+      {
+        :name      => "clone test",
+        :folder    => folder,
+        :host      => host,
+        :pool      => pool,
+        :snapshot  => snapshot,
+        :datastore => datastore
+      }
     end
 
-    it "returns a folder if one is found" do
-      expect(EmsFolder).to receive(:find_by).and_return(:ems_folder)
-      expect(@vm_prov).to receive(:find_folder).never
-      @vm_prov.dest_folder
-    end
+    # Building the disk relocate spec has to connect to the ems so we need to
+    # stub that out here.
+    before { allow(provision).to receive(:build_disk_relocate_spec) }
```

Comment:
NOTE the other option here is to either `allow_any_instance_of(ExtManagementSystem).to receive(:connect)` or `allow(ExtManagementSystem).to receive(:find).with(ems.id).and_return(ems); allow(ems).to receive(:connect)` and mock the call, but both seemed overkill for a single method